Firestore/Source/Local/FSTLevelDBRemoteDocumentCache.mm

```diff
@@ -43,110 +46,35 @@
 using leveldb::DB;
 using leveldb::Status;
 
-@interface FSTLevelDBRemoteDocumentCache ()
-
-@property(nonatomic, strong, readonly) FSTLocalSerializer *serializer;
-
-@end
-
 @implementation FSTLevelDBRemoteDocumentCache {
-  FSTLevelDB *_db;
+  std::unique_ptr<LevelDbRemoteDocumentCache> _cache;
```

Comment:
Optional: I don't think you need polymorphism here, so `_cache` can probably just be a value type.

Reply:
As discussed, we need a way to call the constructor.